perception/sub8_perception/nodes/follow_orange_pipes.py

```diff
@@ -1,318 +1,269 @@
 #!/usr/bin/env python
 import cv2
 import numpy as np
-import sys
 import rospy
-import rospkg
-import sub8_ros_tools
-import tf
-import os
+from tf.transformations import quaternion_from_euler
 from std_msgs.msg import Header
 from std_srvs.srv import SetBool, SetBoolResponse
-from sub8_msgs.srv import VisionRequestResponse, VisionRequest
+from sub8_msgs.srv import VisionRequestResponse, VisionRequest, VisionRequest2D, VisionRequest2DResponse, VisionRequest, VisionRequestResponse
 from geometry_msgs.msg import PoseStamped
-from sub8_vision_tools import MarkerOccGrid, rviz, machine_learning
-from sub8_ros_tools import numpy_quat_pair_to_pose
+from sub8_ros_tools import numpy_quat_pair_to_pose, numpy_to_point, numpy_matrix_to_quaternion, numpy_to_quaternion
+from sub8_ros_tools import Image_Publisher, Image_Subscriber
 from image_geometry import PinholeCameraModel
-
-SEARCH_DEPTH = .65  # m
-
-rospack = rospkg.RosPack()
-
-class MarkerFinder():
+from visualization_msgs.msg import Marker
+
+__author__ = "Kevin Allen"
+
+class PathMarkerFinder():
+    """
+    Node which finds orange path markers in image frame, 
+    estimates the 2d and 3d position/orientation of marker
+    and returns this estimate when service is called.
+    
+    Unit tests for this node is in test_path_marker.py
+    
+    Finding the marker works as follows:
+    * blur image
+    * threshold image mostly for highly saturated, orange/yellow/red objects
+    * run canny edge detection on thresholded image
+    * find contours on edge frame
+    * filter contours to find those that may be contours by:
+      * checking # of sides in appox polygon
+      * checking ratio of length/width close to known model
+    * estimates 3D pose using cv2.solvePnP with known object demensions and camera model
+    
+    TODO: Implement Kalman filter to smooth pose estimate / eliminate outliers
+    """
+    # Model of four corners of path marker, centered around 0 in meters
+    PATH_MARKER = np.array([[0.6096,  -0.0762, 0],
```

Comment:
This should be a ros param. You could also replace the `marker.scale.*` magic constants in the rviz pub function with an expression using it.